### ISSUE TITLE
DatePicker: aria-live not announcing new months

### ISSUE THREAD

**Issue by mtpetros** [adobe/react-spectrum#3559]
# 🐛 Bug Report 

When clicking on the right or left chevron, there should be an aria-live announcement that announces the newly displayed month. This is what occurs when opening the picker initially and reading through; however after you close and reopen the picker, it will no longer announce when clicking the chevron. If you remount the component by refreshing the page,  it will work again as described above. I'm using VoiceOver on Mac in Chrome and in Safari.

## 🤔 Expected Behavior

Screen reader should announce new month when clicking picker chevrons.

## 😯 Current Behavior

Screen reader will only announce new month when clicking chevrons the first time the picker is opened.  If you close and reopen the picker, clicking chevron no longer triggers announcement. 

## 🔦 Context

Our team can't use this component due to this a11y issue.

## 🌍 Your Environment

Spectrum 3.21.2

**Comment by snowystinger** [comment]
I'm unable to reproduce it with MacOS Safari 16 and VO. Please note, Chrome does not have great support for VO and is not a supported combination.

Can you provide an example codesandbox reproducing?

I tested it here https://react-spectrum.adobe.com/react-aria/useDatePicker.html#value and here https://reactspectrum.blob.core.windows.net/reactspectrum/892d41e82dc781fb4651455d0e29c324376659ed/storybook/index.html?path=/story/date-and-time-datepicker--default (our latest build on main)

**Comment by mtpetros** [comment]
Thanks,  @snowystinger. Sorry for the lack of details. So, I was testing react-spectrum DatePicker at https://react-spectrum.adobe.com/react-spectrum/DatePicker.html, which implements useDatePicker. I tested the DatePicker at both the links you posted and I had no issues. But for the react-spectrum DP, I am still having the same issues described above using VO in Safari.

In addition the a11y engineers on our team tested the DP at the same link I provided using NVDA (Chrome + Windows) and ran into the same issue.

**Comment by mtpetros** [comment]
Sorry. Didn't mean to close. 

Thanks,  @snowystinger. Sorry for the lack of details. So, I was testing react-spectrum DatePicker at https://react-spectrum.adobe.com/react-spectrum/DatePicker.html, which implements useDatePicker. I tested the DatePicker at both the links you posted and I had no issues. But for the react-spectrum DP, I am still having the same issues described above using VO in Safari.

In addition the a11y engineers on our team tested the DP at the same link I provided using NVDA (Chrome + Windows) and ran into the same issue.

Thanks!

**Comment by majornista** [comment]
I can reproduce the issue and I think I understand what is happening. When you first open the Calendar overlay in the [DatePicker Example](https://react-spectrum.adobe.com/react-spectrum/DatePicker.html#example), the LiveAnnouncer will not have initialized, because there has never been a need to announce anything. When you then navigate between months, a LiveAnnouncer region gets added at the start of the DOM, and is explicitly excluded from the elements to be hidden by `ariaHideOutside` (see [ariaHideOutside.ts#L33-L36](https://github.com/adobe/react-spectrum/blob/892d41e82dc781fb4651455d0e29c324376659ed/packages/%40react-aria/overlays/src/ariaHideOutside.ts#L33-L36) and [ariaHideOutside.ts#L88-L98](https://github.com/adobe/react-spectrum/blob/892d41e82dc781fb4651455d0e29c324376659ed/packages/%40react-aria/overlays/src/ariaHideOutside.ts#L88-L98)), which allows the live region to announce. However, when the Calendar overlay gets opened after the LiveAnnouncer has initialized, the [`aria-mocal-polyfill`](https://github.com/adobe/react-spectrum/tree/main/packages/%40react-aria/aria-modal-polyfill), used to hide elements outside of the React-Spectrum provider, fails to omit the LiveAnnouncer region from the elements to be hidden from assistive technology. LiveAnnouncer fails to announce because it has been marked as `aria-hidden="true"`.

**Comment by mtpetros** [comment]
This makes sense and explains why after falling to announce the new month(s), if you close the Calendar overlay, presumably at that point `aria-hidden` is set to `true`, and suddenly all the months that failed to announce as you flipped through the calendar are suddenly announced.
